CMakeLists.txt

```diff
@@ -0,0 +1,228 @@
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-multichar -Wpointer-arith -Wshadow -Wnon-virtual-dtor")
+endif()
+
+
+# header detection, could get rid of it, but it is used by lua as well
+include(CheckIncludeFile)
+function(check_has_header FILE_NAME PP_NAME)
+  check_include_file(${FILE_NAME} ${PP_NAME})
+  if(${PP_NAME})
+    add_definitions(-D${PP_NAME}=1)
+  endif()
+endfunction()
+
+check_has_header("stdint.h"    HAVE_STDINT_H)
+check_has_header("inttypes.h"  HAVE_INTTYPES_H)
+check_has_header("sys/types.h" HAVE_SYS_TYPES_H)
+
+
+# endianess detection, could be replaced by including Boost.Config 
+include(TestBigEndian)
+test_big_endian(XOREOS_BIG_ENDIAN)
+if(XOREOS_BIG_ENDIAN)
+  add_definitions(-DXOREOS_BIG_ENDIAN=1)
+else()
+  add_definitions(-DXOREOS_LITTLE_ENDIAN=1)
+endif()
+
+
+# -------------------------------------------------------------------------
+# subfolders where built binaries and libraries will be located, relative to the build folder
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+
+
+# -------------------------------------------------------------------------
+# find the required libraries
+set(XOREOS_LIBRARIES "")
+
+find_package(ZLIB REQUIRED)
+include_directories(${ZLIB_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${ZLIB_LIBRARIES})
+
+find_package(Boost COMPONENTS filesystem system regex date_time REQUIRED)
+include_directories(${Boost_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${Boost_LIBRARIES})
+
+find_package(SDL2 REQUIRED)
+include_directories(${SDL2_INCLUDE_DIR})
+list(APPEND XOREOS_LIBRARIES ${SDL2_LIBRARY})
+
+find_package(OpenAL REQUIRED)
+include_directories(${OPENAL_INCLUDE_DIR})
+list(APPEND XOREOS_LIBRARIES ${OPENAL_LIBRARY})
+
+find_package(OpenGL REQUIRED)
+include_directories(${OPENGL_INCLUDE_DIR})
+list(APPEND XOREOS_LIBRARIES ${OPENGL_LIBRARIES})
+
+find_package(Freetype REQUIRED)
+include_directories(${FREETYPE_INCLUDE_DIRS})
+link_directories(${FREETYPE_LIBRARY_DIRS})
+list(APPEND XOREOS_LIBRARIES ${FREETYPE_LIBRARIES})
+
+find_package(Mad REQUIRED)
+include_directories(${MAD_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${MAD_LIBRARIES})
+
+find_package(Faad REQUIRED)
+include_directories(${FAAD_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${FAAD_LIBRARIES})
+
+find_package(Vorbis REQUIRED)
+include_directories(${VORBIS_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${VORBIS_LIBRARIES})
+
+find_package(XviD REQUIRED)
+include_directories(${XVID_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${XVID_LIBRARIES})
+
+
+find_package(GLEW)
+if(NOT GLEW_FOUND)
+  add_definitions(-DGLEW_STATIC)
+  set(GLEW_LIBRARIES    glew)
+  set(GLEW_INCLUDE_DIRS glew)
+
+  parse_configure(configure.ac glew)
+endif()
+include_directories(${GLEW_INCLUDE_DIRS})
+list(APPEND XOREOS_LIBRARIES ${GLEW_LIBRARIES})
+
+
+find_package(Lua51)
```

Comment:
I'm not sure we actually *ever* want a system Lua instead of our own. We have to support the Witcher scripts, which are compiled, and therefore pinned to a specific Lua version. And we will probably need to hack the loader anyway, to do icky stuff to the bytecode (endianess changes, for one thing).